### PR TITLE
[1/Y] Change server to return hard-coded comments data and display data to the page.

### PR DESCRIPTION
**Pull request by nikhitashanker**
In this change, we make the server return hard-coded comments data. We use Javascript to fetch this data and add it to a HTML element on the page.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -14,19 +14,47 @@
...
+    // Convert the comments into JSON.
+    String json = convertToJson(comments);
+
+    // Send the JSON as the response.
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  /*
+   * Convert List instance into a JSON using the gson library.
```
Comment (hlz03) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:53]: nit: Converts

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -14,19 +14,47 @@
...
+   */
+  private String convertToJson(List<String> messages){
+    Gson gson = new Gson();
+    String json = gson.toJson(messages);
+    return json;
   }
```
Comment (hlz03) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:59]: can just be return gson.toJson() if json is not used other than return.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (nikhitashanker) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (nikhitashanker) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:58]: Done.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -14,19 +14,46 @@
...
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that handles comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private List<String> comments;
+
+  @Override
+  public void init() {
```
Comment (liangqigong) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:33]: I think you don't need this. Check my comment above.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -14,19 +14,46 @@
...
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that handles comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private List<String> comments;
```
Comment (liangqigong) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:30]: I think you can assign the value here. Using comments = ImmutableList.of("","","");

BTW, try to use immutableList whenever possible.

Comment (nikhitashanker) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done. I will try to use ImmutableList whenever possible in the future, thanks.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (nikhitashanker) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Ack.

**Pull request merged**